### PR TITLE
fix: pages configuration

### LINKED ISSUE #10289: Pages configuration doesn't work in NextAuth V5
### Environment

```
System:
  OS: macOS 14.4
  CPU: (8) arm64 Apple M2
  Memory: 1.59 GB / 16.00 GB
  Shell: 5.9 - /bin/zsh
Binaries:
  Node: 18.19.0 - ~/Library/Caches/fnm_multishells/16910_1710223396800/bin/node
  Yarn: 1.22.21 - ~/Library/Caches/fnm_multishells/16910_1710223396800/bin/yarn
  npm: 10.2.3 - ~/Library/Caches/fnm_multishells/16910_1710223396800/bin/npm
  pnpm: 8.14.0 - ~/Library/Caches/fnm_multishells/16910_1710223396800/bin/pnpm
  bun: 1.0.30 - /opt/homebrew/bin/bun
Browsers:
  Safari: 17.4
npmPackages:
  next: canary => 14.2.0-canary.16 
  next-auth: 5.0.0-beta.15 => 5.0.0-beta.15 
  react: ^18.2.0 => 18.2.0 
```

### Reproduction URL

https://github.com/chungweileong94/next-auth-custom-pages-bug

### Describe the issue

In the latest NextAuth V5, if we try to customize the pages via [`pages` options](https://authjs.dev/reference/nextjs#pages), NextAuth will not redirect to the custom page accordingly.

For example: If we set the error page to `/auth/error`, NextAuth will always redirect to the incorrect URL `http://localhost:3000/api/auth/auth/error`.

### How to reproduce

1. Clone the repro code
2. Start the server
3. Navigate to `http://localhost:3000`
4. Click the "Sign In" link
5. Click "Sign In with Credentials", the `CredentialsProvider` should throw a fake error at this point.

### Expected behavior

**Expected**

- It should redirect to `http://localhost:3000/custom/error?error=Configuration`.

**Actual** 
- It redirects to `http://localhost:3000/api/auth/custom/error?error=Configuration` instead.

### PR DESCRIPTION
**Pull request by chungweileong94**
## ☕️ Reasoning

Currently, in the latest next-auth v5, if you customize the error page to `/auth/error` in the `pages` config, next-auth will redirect to `http://localhost:3000/api/auth/auth/error` instead of `http://localhost:3000/auth/error`

## 🧢 Checklist

- [ ] Documentation
- [x] Tests
- [ ] Ready to be merged

## 🎫 Affected issues

Fixes: #10289

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: packages/core/src/index.ts [hunk 1]
```diff
@@ -183,8 +183,9 @@ export async function Auth(
     if (error instanceof CredentialsSignin) params.set("code", error.code)
 
     const pageKind = (isAuthError && error.kind) || "error"
-    const pagePath = config.pages?.[pageKind] ?? `/${pageKind.toLowerCase()}`
-    const url = `${internalRequest.url.origin}${config.basePath}${pagePath}?${params}`
```
Comment (chungweileong94) [inline packages/core/src/index.ts:187]: If we customize the `pages`, it should ignore the `basePath` (`/api/auth`) as well.

**Comment by ThangHuuVu** [comment]
could you add a test for it @chungweileong94 ? 🙏 otherwise LGTM

**Comment by chungweileong94** [comment]
@ThangHuuVu Just added the test in `callback.test.ts`, not sure if this is the right place to put my test. 
Sorry that I was AFK for the weekend.

**Comment by maytees** [comment]
Is there a temporary work around to this in the meantime by any chance? If not, when will the fix be released?

**Comment by chungweileong94** [comment]
> Is there a temporary work around to this in the meantime by any chance? If not, when will the fix be released?

I'm currently using `pnpm patch` to patch the package directly.

**Review by ThangHuuVu — APPROVED** [review]

**Pull request merged**